File: chat/src/main/kotlin/chatox/chat/model/ChatBlocking.kt
package chatox.chat.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime

@Document(collection = "chatBlocking")
data class ChatBlocking(
        @Id
        val id: String,
        @Indexed
        val chatId: String,

        @Indexed
        val blockedUserId: String,

        @Indexed
        val blockedById: String,
        val blockedUntil: ZonedDateTime,
        val description: String?,
        val createdAt: ZonedDateTime,
        val canceled: Boolean,
        val canceledAt: ZonedDateTime?,

        @Indexed
        val canceledById: String? = null,
        val lastModifiedAt: ZonedDateTime?,

        @Indexed
        val lastModifiedById: String? = null
)

File: client/src/api/types/response/ChatBlocking.ts
import {User} from "./User";

export interface ChatBlocking {
    id: string,
    blockedUser: User,
    blockedBy: User,
    blockedUntil: string,
    description?: string,
    canceled: boolean,
    canceledAt?: string,
    canceledBy?: User,
    lastModifiedAt?: string,
    lastModifiedBy?: User,
    chatId: string
}
